pkg/oc/admin/router/router.go

```diff
@@ -229,6 +235,19 @@ type RouterConfig struct {
 	StrictSNI bool
 
 	Local bool
+
+	// MutualTLSAuth controls access to the router using a mutually agreed
+	// upon TLS authentication mechanism (ala client certificates).
+	// One of: required | optional | none  - the default is none.
+	MutualTLSAuth string
+
+	// MutualTLSAuthCA contains the CA certificates that will be used
+	// to verify a client's certificate.
+	MutualTLSAuthCA string
+
+	// MutualTLSAuthCRL contains the certificate revocation list used to
+	// verify a client's certificate.
+	MutualTLSAuthCRL string
 }
```

Comment:
Given we default to no mutual TLS, it may be nice to specify a field with the Subject the cert need to match so that we do not need custom templates but a simple configuration change.

Reply:
Hmm, ok - let me check with @knobunc on today's call on that. 
Reason is that I felt the `ROUTER_MUTUAL_TLS_AUTH_CN` might not even be needed. 
For the specific customer use case, they need to add/remove custom headers, so they would need a custom template to filter on the CN and then do the custom header magic anyway.

Reply:
Hmm, am not sure doing this based on the subject may be possible. I did just briefly glance at the haproxy docs (ref: https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#7.3.4) and didn't see a way to get at the subject field easily.

Reply:
The subject is in SSL_CLIENT_S_DN

Reply:
Or ssl_c_s_dn in haproxy

Reply:
Okay, will rework this a bit later this afternoon to pass that filtering value as a command line option. And to do a substring match on that field `ssl_c_s_dn`. Thx

Reply:
you probably want an exact match

Reply:
so the existing RFE from `https://bugzilla.redhat.com/show_bug.cgi?id=1477349` which the customer has uses a substring match. 
In general, an exact match will be too restrictive if for example you want to filter on just say organizations, departments/units in an organization etc.